pkgs/dkg/drand.go

```diff
@@ -102,44 +101,47 @@ type LocalOwner struct {
 	VerifyFunc func(id uint64, msg, sig []byte) error
 	SignFunc   func([]byte) ([]byte, error)
 
-	Owner common.Address
-	Nonce uint64
-	done  chan struct{}
+	Owner              common.Address
+	Nonce              uint64
+	done               chan struct{}
+	InitiatorPublicKey *rsa.PublicKey
 }
 
 type OwnerOpts struct {
-	Logger     *logrus.Entry
-	ID         uint64
-	BroadcastF func([]byte) error
-	Suite      pairing.Suite
-	VerifyFunc func(id uint64, msg, sig []byte) error
-	SignFunc   func([]byte) ([]byte, error)
-	OpPrivKey  *rsa.PrivateKey
-	Owner      [20]byte
-	Nonce      uint64
+	Logger             *zap.Logger
+	ID                 uint64
+	BroadcastF         func([]byte) error
+	Suite              pairing.Suite
+	VerifyFunc         func(id uint64, msg, sig []byte) error
+	SignFunc           func([]byte) ([]byte, error)
+	OpPrivKey          *rsa.PrivateKey
```

Comment:
If we pass in SignFunc we don't need OpPrivKey, or vice verca, at first I made it a point not to pass in the key and just use the sign function.

Reply:
we use it also to encrypt the share 
https://github.com/bloxapp/ssv-dkg/blob/chores_grooming/pkgs/dkg/drand.go#L250

Reply:
I can add EncryptFunc instead?

Reply:
Added EncryptFunc, DecryptFunc, and rsa pub instead of holding private at each instance